documentation/bolt_running_scripts.md

```diff
@@ -0,0 +1,171 @@
+  loaded from is the `scripts/` directory, which is referenced with `<module>/scripts/myscript.sh`.
+- Use a relative path from the root of your [Bolt project](projects.md).
+- Use an absolute filepath.
+
+Using a Puppet file reference is preferred for a few reasons:
+- It works using the [run_script plan function](./plan_functions.md#run_script), which might be shared
+  with users on other machines that don't have the same file structure
+- It works in other Bolt runners, like Puppet Enterprise
+- It isn't unique to your current system, which makes documentation and plans shareable
+
+To execute a script you'll want to specify:
+- The script file reference (Puppet file ref, relative path, or absolute path)
+- Any arguments the script takes
+- [Bolt CLI options](bolt_command_reference.md#script-run) 
+
+For example, for a script named `update_images.sh` in a module named `manage_docker` on the
+modulepath. The script gets an updated image for every Docker container running in a
+`docker-compose` project:
+
+_update\_images.sh_
+```
+#!/bin/sh
+for c in `docker-compose ps --services |sort`; do
+  echo "redoing $c"
+  docker-compose rm -s -f $c
+  docker-compose pull $c
+done
+docker-compose up -d --remove-orphans
+```
+
+You can run the script with:
+
+_\*nix shell command_
+```shell
+bolt script run manage_docker/scripts/update_images.sh -t <targets>
+```
+
+_PowerShell cmdlet_
+```powershell
+Invoke-BoltScript -Script manage_docker/scripts/update_images.sh -Targets <targets>
+```
+
+## Passing arguments as environment variables
+
+Say the script above is modified to read a `DOCKER_COMPOSE_DIRECTORY` environment variable and cd
+into the value:
+
+_update\_images.sh_
+```
+#!/bin/sh
+if [ -n $DOCKER_COMPOSE_DIRECTORY ]
+  cd $DOCKER_COMPOSE_DIRECTORY
+fi
+
+for c in `docker-compose ps --services |sort`; do
+  echo "redoing $c"
+  docker-compose rm -s -f $c
+  docker-compose pull $c
+done
+docker-compose up -d --remove-orphans
+```
+
+You can run the script with:
+
+_\*nix shell command_
+```shell
+bolt script run manage_docker/scripts/update_images.sh -t <targets> --env-var DOCKER_COMPOSE_DIRECTORY=./docker
+```
+
+_PowerShell cmdlet_
+```powershell
+Invoke-BoltScript -Script manage_docker/scripts/update_images.sh -Targets <targets> -EnvVar DOCKER_COMPOSE_DIRECTORY=.\docker
+```
+
+## Passing arguments on the command line
+
+If your script accepts command-line arguments, like this:
+
+_update\_images.sh_
+```
+#!/bin/sh
+cd $1
+for c in `docker-compose ps --services |sort`; do
+  echo "redoing $c"
+  docker-compose rm -s -f $c
+  docker-compose pull $c
+done
+docker-compose up -d --remove-orphans
+```
+
+You can run the script with:
+
+_\*nix shell command_
+```shell
+bolt script run manage_docker/scripts/update_images.sh ./docker -t <targets>
+```
+
+_PowerShell cmdlet_
+```powershell
+Invoke-BoltScript -Script manage_docker/scripts/update_images.sh .\docker -Targets <targets>
+```
+
+## Passing PowerShell parameters
```

Comment:
Unless I'm missing something, I don't think the script and command described in this section work as shown. PowerShell won't do any type coercion when you pass parameters like this. Here's a simplified example:

```powershell
[CmdletBinding()]
Param(
  [Int]$Timeout = 3,
  [Boolean]$Shutdown = $false
)

Write-Output "timeout: $timeout ; shutdown_only: $shutdown_only"
```

```shell
$ bolt script run ~/.puppetlabs/bolt/scripts/test.ps1 "\-Timeout 30 \-Shutdown \$true" -t windows

Started on win...
Failed on win:
  The command failed with exit code 1
  [InvalidCastFromStringToInteger,Microsoft.PowerShell.Commands.InvokeCommandCommand] Exception At line:33 char:3
  +   Invoke-Command @invokeArgs
  +   ~~~~~~~~~~~~~~~~~~~~~~~~~~.
  Cannot convert value "\-Timeout 30 \-Shutdown true" to type "System.Int32". Error: "Input string was not in a correct format."
Failed on 1 target: win
Ran on 1 target in 2.16 sec
```

You _can_ pass them as positional arguments, though, and PowerShell will assign them to the parameters correctly:

```shell
$ bolt script run ~/.puppetlabs/bolt/scripts/test.ps1 30 true -t windows

Started on win...
skip_cleanup [ID: Skipping cleanup of tmpdir 'C:\Users\Administrator\AppData\Local\Temp\oinbialq.ogy']
Finished on win:
  timeout: 30 ; shutdown_only: True
Successful on 1 target: win
Ran on 1 target in 2.13 sec
```

Reply:
Gah, I forgot scripts don't support named arguments. Just passing `30 true` works though. Updated.